src/main/resources/additional_properties.json

```diff
@@ -18,5 +18,266 @@
         "$ref" : "#/$defs/serializer"
       }
     }
+  },
+  "processors" : {
+    "type" : "array",
+    "items" : {
+      "oneOf" : [
+        {
+          "type" : "object",
+          "required" : [ "cast_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "cast_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "spec"],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "spec" : {
+                  "type" : "array",
+                  "items" : {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "insert_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "insert_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field", "value" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "string"
+                },
+                "value" : {
+                  "type" : "string"
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "extract_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "extract_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "string"
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "mask_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "mask_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "replace_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "replace_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "whitelist", "blacklist", "renames" ],
```

Comment:
We don't want to use divisive language like "whitelist" and "blacklist" anymore.
Furthermore I think replace field has multiple very different use-cases: exclude list/dropping fields, include list and renaming. What do you think we make 3 different processors out of that?